austation/code/modules/mining/minebot.dm

```diff
@@ -0,0 +1,95 @@
+#define MINEDRONE_COLLECT 1
+#define MINEDRONE_ATTACK 2
+
+/mob/living/simple_animal/hostile/mining_drone
+	var/beacons = 15
+	var/default_hatmask
+	var/obj/item/t_scanner/adv_mining_scanner/lesser/scanner //there is no code to turn this scanner off.
+
+/mob/living/simple_animal/hostile/mining_drone/Initialize()
+	. = ..()
+	weather_immunities += "ash" // no damage from ash storms
```

Comment:
set the weather immunities as an object variable, not in the Init proc. Only reason they have to be changed at runtime is for effects or upgrades. If it's set with a simple value in init, it should be an object var.

Reply:
Yes I know you can't do operations with object vars, but all this does is add an item to an empty list.
Add an object var of `weather_immunities = list("ash")`

Reply:
okay.
I did it this way in case Bee ever changes this variable.

Reply:
now defined in class